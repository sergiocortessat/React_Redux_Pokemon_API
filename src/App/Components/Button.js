import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ type, handleClick }) => {
  const style = {
    backgroundColor: type.color,
    color: type.text,
  };
  return (
    <button
      type="button"
      className="type-filter-item"
      style={style}
      data-testid="button"
      onClick={(e) => handleClick(e.target.innerText)}
    >
      {type.name}
      {' '}
    </button>
  );
};
Button.defaultProps = {
  type: {
    name: '',
    color: '',
    text: '',
  },
  handleClick: () => {},
};

Button.propTypes = {
  type: PropTypes.objectOf(PropTypes.string),
  handleClick: PropTypes.func,
};

export default Button;
