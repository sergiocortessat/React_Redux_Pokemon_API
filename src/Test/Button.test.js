import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Button from '../App/Components/Button';

describe('Button to sort', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Button />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
