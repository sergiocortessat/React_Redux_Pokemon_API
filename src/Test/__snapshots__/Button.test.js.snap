// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button to sort should match with snapshot 1`] = `
<button
  className="type-filter-item"
  data-testid="button"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "",
      "color": "",
    }
  }
  type="button"
>
  
   
</button>
`;
